pkg/xds/hooks/resource_set.go

```diff
@@ -9,6 +9,10 @@ import (
 // Since resourceSet is an argument, you can add new, remove or modify the existing resources
 // To support V2 and V3, search the resource set for typeUrls for proper version
 // If you want to for example modify only inbound listeners, search for the resource origin of OriginInbound
+//
+// WARNING: Please, be aware that Hooks are being called also in Ingress Dataplanes
+//			and when method Modify is being called, the passed xds_context.Context
```

Comment:
xds_context.Context will not be empty. MeshContext inside will be empty

Reply:
changed